spec.emu

```diff
@@ -4,36 +4,421 @@
+    <p>It performs the following steps when called:</p>
+    <emu-alg>
+      1. Let _O_ be the *this* value.
+      1. Perform ? RequireInternalSlot(_O_, [[AmountData]]).
+      1. Perform ? RequireInternalSlot(_O_, [[FractionDigits]]).
+      1. If _options_ is not *undefined*, then
+        1. If _options_ is not an Object, throw a *TypeError* exception.
+      1. Let _v_ be _O_.[[AmountData]].
+      1. Let _numDigits_ be _O_.[[FractionDigits]].
+      1. Assert: _numDigits_ is a non-negative integer.
+      1. Let _roundingMode_ be *"halfEven"*.
+      1. If _options_ is not *undefined*, then
+        1. Let _roundingModeFromOpts_ be ? Get(_options_, *"roundingMode"*).
+        1. If _roundingModeFromOpts_ is not *undefined*, then
+          1. If _roundingModeFromOpts_ is not a String, throw a *TypeError* exception.
+          1. If _roundingModeFromOpts_ is not a rounding mode, throw a *RangeError* exception.
+          1. Set _roundingMode_ to _roundingModeFromOpts_.
+      1. Let _s_ be RenderMVWithFractionDigits(_v_, _numDigits_, _roundingMode_).
+      1. If _O_.[[Unit]] is *undefined*, return _s_.
+      1. Otherwise, return the string concatenation of _s_, *"["*, _O_.[[Unit]], and *"]"*.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="sec-amount.prototype.tolocalestring">
+    <h1>Amount.prototype.toLocaleString ( [ _reserved1_ [ , _reserved2_ ] ] )</h1>
+      <p>An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement this method as specified in the ECMA-402 specification. If an ECMAScript implementation does not include the ECMA-402 API the following specification of this method is used:</p>
+      <p>This method produces a String value that represents this Amount object formatted according to the conventions of the host environment’s current locale. This method is implementation-defined, and it is permissible, but not encouraged, for it to return the same thing as <emu-xref href="#sec-amount.prototype.tostring">`toString`</emu-xref>.</p>
+      <p>The meanings of the optional parameters to this method are defined in the ECMA-402 specification; implementations that do not include ECMA-402 support must not use those parameter positions for anything else.</p>
+  </emu-clause>
+
+  <emu-clause id="sec-amount.prototype.with">
+    <h1>Amount.prototype.with ( _options_ )</h1>
+    <p>This method returns a fresh Amount object whose underlying mathematical value is equal to the mathematical value of the current Amount, but possibly with different precision as specified in the _options_.</p>
+    <p>It performs the following steps when called:</p>
+    <emu-alg>
+      1. Let _O_ be the *this* value.
+      1. Perform ? RequireInternalSlot(_O_, [[AmountData]]).
+      1. If _options_ is not an Object, throw a *TypeError* exception.
+      1. Let _v_ be _O_.[[AmountData]].
+      1. Let _s_ be Number::toString(_v_, 10).
+      1. Return ? Call (%Amount%, _O_, « _s_, _options_ »).
```

Comment:
call, or construct?

Reply:
Ah, good spot. I mean construct. The `OrdinaryObjectCreate` is what I wanted.